tests/ui/confusing_xor_and_pow.rs

```diff
@@ -0,0 +1,16 @@
+#![allow(unused)]
+#![warn(clippy::confusing_xor_and_pow)]
+#![allow(clippy::eq_op)]
+
+fn main() {
+    // Should warn:
+    let _ = 2i32 ^ 9i32;
+    let _ = 50i32 ^ 3i32;
+    let _ = 5i32 ^ 8i32;
+    let _ = 2i32 ^ 32i32;
```

Comment:
good to always add a test with a macro since they can mess up the suggestion.

something even as basic as 

```rust
macro_rules! number {
    () => {2 + 3 * 4}
};

let _ = 6i32 ^ number!();
```